### PR TITLE
remove CircleCI, use github actions

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: build-and-test
```
Comment (crobert-1) [inline .github/workflows/test.yml:1]: Would it make sense to have the file name match the name here?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  checklinks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check
```
Comment (crobert-1) [inline .github/workflows/test.yml:13]: Maybe make the name "Check links", like the job title above? Probably not a big deal, I just like a little more descriptive titles when possible.

**Review by crobert-1 — APPROVED** [review]

**Comment by crobert-1** [comment]
I know I'm late, just little nits anyways :)

**Comment by atoulme** [comment]
I can create a follow up.

**Comment by atoulme** [comment]
Here you go: https://github.com/signalfx/integrations/pull/997

**Pull request merged**